Test-№2.1-from-14.12.2022/task1.c

```diff
@@ -0,0 +1,197 @@
+#include <stdio.h>
+#include <stdbool.h>
+
+void reset(bool *existCheck)
+{
+    for (int i = 0; i < 3; ++i)
+    {
+        existCheck[i] = false;
+    }
+}
+
+void resetTemp(int *array)
+{
+    for (int i = 0; i < 8; ++i)
+    {
+        array[i] = 0;
+    }
+}
+
+int fastPower(int a, int b) // a^b
+{
+    if (b == 0)
+    {
+        return 1;
+    } else if (b % 2)
+    {
+        return fastPower(a, b / 2) * fastPower(a, b / 2) * a;
+    } else
+    {
+        return fastPower(a, b / 2) * fastPower(a, b / 2);
+    }
+}
+
+bool isFileTabulationCorrect(FILE *file)
+{
+    if (file == NULL)
+    {
+        return false;
+    }
+    bool existCheck[3] = {false, false, false};
+    // 0 - itemNameExist = false;
+    // 1 - itemCntExist = false;
+    // 2 - itemPriceExist = false;
+    char c = fgetc(file);
+    int pointer = 0;
+    while (c != EOF)
+    {
+        if ((c != '\t') && (c != '\n'))
+        {
+            existCheck[pointer] = true;
+        } else if (c == '\t')
+        {
+            ++pointer;
+            if (pointer >= 3)
+            {
+                return false;
+            }
+        } else
+        {
+            pointer = 0;
+            if (!(existCheck[0] && existCheck[1] && existCheck[2]))
+            {
+                return false;
+            }
+            reset(existCheck);
+        }
+        c = fgetc(file);
+    }
+    if (!(existCheck[0] && existCheck[1] && existCheck[2]))
+    {
+        return false;
+    }
+    return true;
+}
+
+int toInt(int *array)
+{
+    int answer = 0;
+    for (int i = 0; i < 8; ++i)
+    {
+        answer += array[i] * fastPower(10, i);
+    }
+    return answer;
+}
+
+long long orderCounter(FILE *file)
+{
+    long long preResult = 0;
+    long long result = 0;
+    int temp[8] = {0};
+    int counterOfItem = 0;
```

Comment:
```suggestion
    int countOfItems = 0;
```
Это не счетчик предмета, а количество предметов